src/utils/trace_parser.py

```diff
@@ -107,23 +117,37 @@ def read_line(self) -> Optional[Tuple[TraceCommand, Optional[int]]]:
         try:
             # Split fields ->> [Operation] [Address]
             parts = line.split()
+
+            try:
+                op = TraceCommand(int(parts[0]))  # Cast to enum class
+            except (ValueError, IndexError):
+                warnings.warn(f"Invalid line format: {line.strip()}, reading next line")
+                return self.read_line()  # Recurse to next line
+
+            # handle RESET and PRINT commands
+            if op in (TraceCommand.CLEAR_CACHE, TraceCommand.PRINT_CACHE):
+                return op, None
```

Comment:
Since this returns `None`, should we just remove [this debug logging](https://github.com/nkanderson/ECE585-llc/blob/7282c6fdfe30d6ad032f089420ca51a99213ea8f/src/main.py#L43-L46) from the `main` program that's throwing an error now?